jax/_src/basearray.py

```diff
@@ -17,30 +17,20 @@
 import abc
 
 class Array(abc.ABC):
-  """Experimental Array base class for JAX
-
-  `jax.Array` is meant as the future public interface for instance checks and
-  type annotation of JAX array objects. JAX Array object types are currently in
-  flux, and this class only fully supports the new `jax.experimental.Array`, which
-  will soon replace the old-style {class}`DeviceArray`, {class}`ShardedDeviceArray`,
-  {class}`GlobalDeviceArray`, etc.
-
-  The compatibility is summarized in the following table:
-
-  ================================  ======================  =========================
-  object type                       ``isinstance`` support  type annotation support
-  ================================  ======================  =========================
-  {class}`DeviceArray`               ✅                      ❌
-  {class}`ShardedDeviceArray`        ✅                      ❌
-  {class}`GlobalDeviceArray`         ✅                      ❌
-  {class}`~jax._src.core.Tracer`     ✅                      ✅
-  {class}`~jax.experimental.Array`   ✅                      ✅
-  ================================  ======================  =========================
-
-  In other words, ``isinstance(x, jax.Array)`` will return True for any of these types,
-  whereas annotations such as ``x : jax.Array`` will only type-check correctly for
-  instances of {class}`~jax._src.core.Tracer` and {class}`jax.experimental.Array`, and
-  not for the other soon-to-be-deprecated array types.
+  """Array base class for JAX
+
+  ``jax.Array`` is the public interface for instance checks and type annotation of JAX
+  arrays and tracers. Its main applications are in instance checks and type annotations;
+  for example::
+
+    x = jnp.arange(5)
+    isinstance(x, jax.Array)  # returns True both inside and outside traced functions.
+
+    def f(x: Array) -> Array:  # type annotations are valid for traced and non-traced types.
+      return x
+
+  Because `jax.Array` is effectively an abstact base class, the main documentation for
```

Comment:
sp. abstract

I'm actually a bit surprised to not find a list of methods, at least, even if they link to the numpy functions.

I went looking on the `jax.numpy` docs and didn't really find what I wanted there either.

Reply:
Yeah, we need to figure out how to get a list of array methods in the HTML docs; there was a chat thread on that this morning. I'm going to work on that in a followup.